sklearn/metrics/regression.py

```diff
@@ -535,6 +538,12 @@ def r2_score(y_true, y_pred, sample_weight=None,
         y_true, y_pred, multioutput)
     check_consistent_length(y_true, y_pred, sample_weight)
 
+    if _num_samples(y_pred) < 2:
+        msg = "Found an array of {0} sample(s), while the minimum number is 2."
```

Comment:
Or maybe "R^2 score is not well-defined with less than two samples!", not sure though.